### PR TITLE
New Resource: `azurerm_application_load_balancer_frontend`

### PR DESCRIPTION
**Pull request by ziyeqf**
`azurerm_application_load_balancer` is also affected, and also tested.

test
---
```
❯ tftest servicenetworking TestAccApplicationLoadBalancer
=== RUN   TestAccApplicationLoadBalancerFrontend_basic
=== PAUSE TestAccApplicationLoadBalancerFrontend_basic
=== RUN   TestAccApplicationLoadBalancerFrontend_complete
=== PAUSE TestAccApplicationLoadBalancerFrontend_complete
=== RUN   TestAccApplicationLoadBalancerFrontend_update
=== PAUSE TestAccApplicationLoadBalancerFrontend_update
=== RUN   TestAccApplicationLoadBalancerFrontend_requiresImport
=== PAUSE TestAccApplicationLoadBalancerFrontend_requiresImport
=== RUN   TestAccApplicationLoadBalancer_basic
=== PAUSE TestAccApplicationLoadBalancer_basic
=== RUN   TestAccApplicationLoadBalancer_complete
=== PAUSE TestAccApplicationLoadBalancer_complete
=== RUN   TestAccApplicationLoadBalancer_update
=== PAUSE TestAccApplicationLoadBalancer_update
=== RUN   TestAccApplicationLoadBalancer_requiresImport
=== PAUSE TestAccApplicationLoadBalancer_requiresImport
=== CONT  TestAccApplicationLoadBalancerFrontend_basic
=== CONT  TestAccApplicationLoadBalancer_basic
=== CONT  TestAccApplicationLoadBalancer_update
=== CONT  TestAccApplicationLoadBalancer_requiresImport
=== CONT  TestAccApplicationLoadBalancer_complete
=== CONT  TestAccApplicationLoadBalancerFrontend_update
=== CONT  TestAccApplicationLoadBalancerFrontend_requiresImport
=== CONT  TestAccApplicationLoadBalancerFrontend_complete
--- PASS: TestAccApplicationLoadBalancer_requiresImport (662.87s)
--- PASS: TestAccApplicationLoadBalancer_basic (675.26s)
--- PASS: TestAccApplicationLoadBalancerFrontend_complete (695.94s)
--- PASS: TestAccApplicationLoadBalancer_update (737.62s)
--- PASS: TestAccApplicationLoadBalancer_complete (751.68s)
--- PASS: TestAccApplicationLoadBalancerFrontend_requiresImport (770.02s)
--- PASS: TestAccApplicationLoadBalancerFrontend_basic (778.00s)
--- PASS: TestAccApplicationLoadBalancerFrontend_update (919.29s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/servicenetworking     919.373s

```

### CODE REVIEW

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @ziyeqf 

Thanks for this PR - I've taken a look through and left some comments inline, but I'm unsure why we've reverted the naming back to as in the previous PR, so can we fix this?

Thanks!

**Comment by ziyeqf** [comment]
new test
---
```
terraform-provider-azurerm [ tengzh/alb/frontend][$⇡] [🐹 v1.21.1]
❯ tftest servicenetworking TestAccApplicationLoadBalancerFrontend
=== RUN   TestAccApplicationLoadBalancerFrontend_basic
=== PAUSE TestAccApplicationLoadBalancerFrontend_basic
=== RUN   TestAccApplicationLoadBalancerFrontend_complete
=== PAUSE TestAccApplicationLoadBalancerFrontend_complete
=== RUN   TestAccApplicationLoadBalancerFrontend_update
=== PAUSE TestAccApplicationLoadBalancerFrontend_update
=== RUN   TestAccApplicationLoadBalancerFrontend_requiresImport
=== PAUSE TestAccApplicationLoadBalancerFrontend_requiresImport
=== CONT  TestAccApplicationLoadBalancerFrontend_basic
=== CONT  TestAccApplicationLoadBalancerFrontend_update
=== CONT  TestAccApplicationLoadBalancerFrontend_requiresImport
=== CONT  TestAccApplicationLoadBalancerFrontend_complete
--- PASS: TestAccApplicationLoadBalancerFrontend_update (742.62s)
--- PASS: TestAccApplicationLoadBalancerFrontend_basic (756.34s)
--- PASS: TestAccApplicationLoadBalancerFrontend_requiresImport (759.31s)
--- PASS: TestAccApplicationLoadBalancerFrontend_complete (812.33s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/servicenetworking     812.387s
```

the CI failure seems caused by some loading reason, I do not have permision to re-trigger it. Could you please re-run it?

File: internal/services/servicenetworking/application_load_balancer_frontend_resource.go [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+			}
+
+			update := frontendsinterface.FrontendUpdate{}
+
+			if metadata.ResourceData.HasChange("tags") {
+				update.Tags = tags.Expand(config.Tags)
+			}
+
+			if _, err := client.Update(ctx, *id, update); err != nil {
+				return fmt.Errorf("updating `azurerm_alb_frontend` %s: %+v", *id, err)
```
Comment (catriona-m) [inline internal/services/servicenetworking/application_load_balancer_frontend_resource.go:194]:
```suggestion
				return fmt.Errorf("updating `azurerm_application_load_balancer_frontend` %s: %+v", *id, err)
```

File: internal/services/servicenetworking/application_load_balancer_frontend_resource.go [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+func (f FrontendsResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.ServiceNetworking.FrontendsInterface
+
+			id, err := frontendsinterface.ParseFrontendID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("parsing %s: %+v", metadata.ResourceData.Id(), err)
```
Comment (catriona-m) [inline internal/services/servicenetworking/application_load_balancer_frontend_resource.go:140]:
```suggestion
				return err
```

**Review by catriona-m — CHANGES_REQUESTED** [review]
Hi @ziyeqf - thanks for fixing this up, I left two minor comments but overall this is looking good! Thanks!

File: internal/services/servicenetworking/application_load_balancer_frontend_resource.go [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding %v", err)
+			}
+
+			update := frontendsinterface.FrontendUpdate{}
+
+			if metadata.ResourceData.HasChange("tags") {
+				update.Tags = tags.Expand(config.Tags)
+			}
+
```
Comment (katbyte) [inline internal/services/servicenetworking/application_load_balancer_frontend_resource.go:192]:
```suggestion

```

File: internal/services/servicenetworking/application_load_balancer_resource.go [hunk 4]
```diff
@@ -186,8 +187,39 @@ func (t ApplicationLoadBalancerResource) Delete() sdk.ResourceFunc {
 			if err != nil {
 				return err
 			}
-			if err := client.DeleteThenPoll(ctx, *id); err != nil {
-				return fmt.Errorf("deleting %s: %+v", *id, err)
+
+			// a workaround for that some child resources may still exist for seconds before it fully deleted.
+			// tracked o https://github.com/Azure/azure-rest-api-specs/issues/26000n
```
Comment (katbyte) [inline internal/services/servicenetworking/application_load_balancer_resource.go:192]:
```suggestion
			// tracked o https://github.com/Azure/azure-rest-api-specs/issues/26000
```

File: internal/services/servicenetworking/application_load_balancer_frontend_resource_test.go [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+func (r ApplicationLoadBalancerFrontendResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {
+  }
+}
+
+	%s
```
Comment (katbyte) [inline internal/services/servicenetworking/application_load_balancer_frontend_resource_test.go:128]:
```suggestion
%s
```

File: internal/services/servicenetworking/application_load_balancer_frontend_resource_test.go [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+}
+
+func (r ApplicationLoadBalancerFrontendResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {
+  }
+}
+
+	%s
```
Comment (katbyte) [inline internal/services/servicenetworking/application_load_balancer_frontend_resource_test.go:144]:
```suggestion
%s
```

**Review by katbyte — APPROVED** [review]
LGTM aside from a couple comments ⛈️

**Review by katbyte — APPROVED** [review]
LGTM aside from a couple comments ⛈️

**Review by katbyte — APPROVED** [review]
LGTM aside from a couple comments ⛈️

**Pull request merged**